core/src/main/java/org/apache/calcite/rel/core/AggregateCall.java

```diff
@@ -292,9 +292,15 @@ public AggregateCall rename(String name) {
   }
 
   public String toString() {
-    StringBuilder buf = new StringBuilder(aggFunction.toString());
+    StringBuilder buf = new StringBuilder();
+    // currently approximate = true is only for 'APPROX_COUNT_DISTINCT'
+    if (approximate && distinct) {
+      buf.append("APPROX_COUNT_DISTINCT");
+    } else {
```

Comment:
Agreed with Julian, we should not modify the agg call name which is a unique identifier of the function, better to add some distinct flag to the Aggregate digest instead:
```java
public RelWriter explainTerms(RelWriter pw) {
    // We skip the "groups" element if it is a singleton of "group".
    super.explainTerms(pw)
        .item("group", groupSet)
        .itemIf("groups", groupSets, getGroupType() != Group.SIMPLE)
        .itemIf("aggs", aggCalls, pw.nest());
    if (!pw.nest()) {
      for (Ord<AggregateCall> ord : Ord.zip(aggCalls)) {
        pw.item(Util.first(ord.e.name, "agg#" + ord.i), ord.e);
      }
    }
    return pw;
  }
```

Reply:
I would prefer to add some distinct flag to AggregateCall rather than  Aggregate,  because  'approximate' is a property of AggregateCall, not Aggregate. What do you think, Danny?

Reply:
Either is fine to me, go with your choice. BTW, that format of the distinct call would be like ? `distinct call_name(args)`  ?